lektor/admin/webui.py

```diff
@@ -52,7 +52,7 @@ def resolve_artifact(self, path, pad=None, redirect_slash=True):
             # and it ensures our relative urls work.
             if not path.endswith('/') and \
                source.url_path != '/' and \
-               source.url_path.endswith('/'):
+               source.url_path != path:
```

Comment:
This change is needed if you have a file "assets/dir/index.html" and visit "localhost:5000/dir".

Before, this resolved directly to "index.html", because "source.url_path" is "assets/dir/index.html", which does not end with "/".

Now, the dev server redirects you from "localhost:5000/dir" to ""localhost:5000/dir/" before it serves you index.html.